### PR TITLE
mqtt: Gives user data in config

### PR DESCRIPTION
**Pull request by ceolin**
It is the same patter used by message digest.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

### CODE REVIEW

File: src/lib/comms/include/sol-mqtt.h [hunk 1]
```diff
@@ -341,11 +346,10 @@ struct sol_mqtt_config {
  * @param host The host address of the MQTT broker
  * @param port The host port to connect to
  * @param config Configuration and callbacks
- * @param data User data provided to the callbacks
  *
  * @return New mqtt object on sucess, NULL otherwise
  */
-struct sol_mqtt *sol_mqtt_connect(const char *host, uint16_t port, const struct sol_mqtt_config *config, const void *data);
+struct sol_mqtt *sol_mqtt_connect(const char *host, uint16_t port, const struct sol_mqtt_config *config);
```
Comment (barbieri) [inline src/lib/comms/include/sol-mqtt.h]: I'm thinking if we should move host/port to the config as well... what do you think?

File: src/lib/comms/sol-mqtt-impl-mosquitto.c [hunk 2]
```diff
@@ -371,7 +371,7 @@ sol_mqtt_connect(const char *host, uint16_t port, const struct sol_mqtt_config *
 
     /* It comes as const, but goes back to user on the callbacks as
      * not const, for convenience */
-    mqtt->data = (void *)data;
+    mqtt->data = (void *)config->data;
```
Comment (barbieri) [inline src/lib/comms/sol-mqtt-impl-mosquitto.c]: take the time to store this as `const` and only cast on callbacks?

File: src/lib/comms/include/sol-mqtt.h (hunk 1 above)
Comment (vcgomes) [inline src/lib/comms/include/sol-mqtt.h]: Hm, now that there's support for naming resolving, we could consider using `sol_network_link_addr` here. Just an idea.

**Comment by vcgomes** [comment]
typo on the commit message: s/patter/pattern/ .

**Comment by cabelitos** [comment]
@ceolin Maybe we should do the same thing for LWM2M.

File: src/lib/comms/include/sol-mqtt.h (hunk 1 above)
Comment (ceolin) [inline src/lib/comms/include/sol-mqtt.h]: i do not think is better, libmosquitto expects a string (host) and an int (port), so it'll be necessary extra convertions if we use `sol_network_link_addr`

File: src/lib/comms/sol-mqtt-impl-mosquitto.c (hunk 2 above)
Comment (ceolin) [inline src/lib/comms/sol-mqtt-impl-mosquitto.c]: yep

File: src/lib/comms/include/sol-mqtt.h (hunk 1 above)
Comment (ceolin) [inline src/lib/comms/include/sol-mqtt.h]: @barbieri this is the connect function, i guess we can keep the address out of the config

**Comment by ceolin** [comment]
https://github.com/solettaproject/soletta/pull/1860

**Pull request closed without merging**